finals-2024-01-15/code/5215-berat-useini/src/app/services/movieServices/movie.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import {map, Observable} from 'rxjs';
+import { Movie } from '../../models/movie.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MovieService {
+  private apiUrl = 'http://localhost:3000/movies';
+
+  constructor(private http: HttpClient) {}
+
+  getMovies(): Observable<Movie[]> {
+    return this.http.get<Movie[]>(this.apiUrl);
+  }
+
+  getGenres(): Observable<string[]> {
+    return this.http.get<string[]>(`${this.apiUrl}/genres`);
```

Comment:
This asks for the wrong url - `http://localhost:3000/movies/genres` instead of `http://localhost:3000/genres`